# Query (Consulta) de dados no DataFrame

In [2]:
import pandas as pd
from pandas import DataFrame

In [3]:
df = pd.read_csv("dataset2.csv")

In [4]:
df.head(5)

ID_Pedido Data_Pedido ID_Cliente   Segmento           Pais Regiao  \
0  CA-2016-152156  2016-11-08   CG-12520   Consumer  United States  South   
1  CA-2016-152156  2016-11-08   CG-12520   Consumer  United States  South   
2  CA-2016-138688  2016-06-12   DV-13045  Corporate  United States   West   
3  US-2015-108966  2015-10-11   SO-20335   Consumer  United States  South   
4  US-2015-108966  2015-10-11   SO-20335   Consumer  United States  South   

        ID_Produto        Categoria  \
0  FUR-BO-10001798        Furniture   
1  FUR-CH-10000454        Furniture   
2  OFF-LA-10000240  Office Supplies   
3  FUR-TA-10000577        Furniture   
4  OFF-ST-10000760  Office Supplies   

                                        Nome_Produto  Valor_Venda  Quantidade  
0                  Bush Somerset Collection Bookcase     261.9600         NaN  
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...     731.9400         NaN  
2  Self-Adhesive Address Labels for Typewriters b...      14.6200         2.0  
3      Bretford CR4500 Series Slim Rectangular Table     957.5775         5.0  
4                     Eldon Fold 'N Roll Cart System      22.3680         2.0

In [5]:
# Checamos os valores mínimo e máximo da coluna Valor_Venda
df.Valor_Venda.describe()

count     9994.000000
mean       229.858001
std        623.245101
min          0.444000
25%         17.280000
50%         54.490000
75%        209.940000
max      22638.480000
Name: Valor_Venda, dtype: float64

O intervalo de vendas (Valor_Venda) é de 0.44 a 22638. Vamos fazer uma consulta e retornar todas as vendas dentro de um range de valores.

In [6]:
# Geramos um novo dataframe apenas com o intervalo de vendas entre 229 e 1000
df2 = df.query('229 < Valor_Venda < 10000')

In [7]:
# Então confirmamos os valores mínimo e máximo 
df2.Valor_Venda.describe()

count    2357.000000
mean      766.679142
std       856.315136
min       229.544000
25%       323.100000
50%       490.320000
75%       859.200000
max      9892.740000
Name: Valor_Venda, dtype: float64

In [8]:
# Geramos um novo dataFrame apenas com os valores de venda acima da média
df3 = df2.query('Valor_Venda > 766')

In [9]:
df3.head()

ID_Pedido Data_Pedido ID_Cliente  Segmento           Pais Regiao  \
3   US-2015-108966  2015-10-11   SO-20335  Consumer  United States  South   
7   CA-2014-115812  2014-06-09   BH-11710  Consumer  United States   West   
10  CA-2014-115812  2014-06-09   BH-11710  Consumer  United States   West   
11  CA-2014-115812  2014-06-09   BH-11710  Consumer  United States   West   
24  CA-2015-106320  2015-09-25   EB-13870  Consumer  United States   West   

         ID_Produto   Categoria  \
3   FUR-TA-10000577   Furniture   
7   TEC-PH-10002275  Technology   
10  FUR-TA-10001539   Furniture   
11  TEC-PH-10002033  Technology   
24  FUR-TA-10000577   Furniture   

                                     Nome_Produto  Valor_Venda  Quantidade  
3   Bretford CR4500 Series Slim Rectangular Table     957.5775         5.0  
7                  Mitel 5320 IP Phone VoIP phone     907.1520         6.0  
10       Chromcraft Rectangular Conference Tables    1706.1840         9.0  
11  Konftel 250 Conference phone - Charcoal black     911.4240         4.0  
24  Bretford CR4500 Series Slim Rectangular Table    1044.6300         3.0

# Ocorrência de Valores em uma Coluna

In [10]:
df.shape

(9994, 11)

In [12]:
# Então aplicamos o filtro
df[df['Quantidade'].isin([5,7,9,11])]

ID_Pedido Data_Pedido ID_Cliente     Segmento           Pais  \
3     US-2015-108966  2015-10-11   SO-20335     Consumer  United States   
5     CA-2014-115812  2014-06-09   BH-11710     Consumer  United States   
9     CA-2014-115812  2014-06-09   BH-11710     Consumer  United States   
10    CA-2014-115812  2014-06-09   BH-11710     Consumer  United States   
14    US-2015-118983  2015-11-22   HP-14815  Home Office  United States   
...              ...         ...        ...          ...            ...   
9974  US-2016-103674  2016-12-06   AP-10720  Home Office  United States   
9977  US-2016-103674  2016-12-06   AP-10720  Home Office  United States   
9981  CA-2017-163566  2017-08-03   TB-21055     Consumer  United States   
9982  US-2016-157728  2016-09-22   RC-19960     Consumer  United States   
9988  CA-2017-163629  2017-11-17   RA-19885    Corporate  United States   

       Regiao       ID_Produto        Categoria  \
3       South  FUR-TA-10000577        Furniture   
5        West  FUR-FU-10001487        Furniture   
9        West  OFF-AP-10002892  Office Supplies   
10       West  FUR-TA-10001539        Furniture   
14    Central  OFF-AP-10002311  Office Supplies   
...       ...              ...              ...   
9974     West  OFF-AR-10004752  Office Supplies   
9977     West  OFF-FA-10003467  Office Supplies   
9981     East  OFF-LA-10004484  Office Supplies   
9982  Central  OFF-PA-10002195  Office Supplies   
9988    South  TEC-PH-10004006       Technology   

                                           Nome_Produto  Valor_Venda  \
3         Bretford CR4500 Series Slim Rectangular Table     957.5775   
5     Eldon Expressions Wood and Plastic Desk Access...      48.8600   
9                      Belkin F5C206VTEL 6 Outlet Surge     114.9000   
10             Chromcraft Rectangular Conference Tables    1706.1840   
14    Holmes Replacement Filter for HEPA Air Cleaner...      68.8100   
...                                                 ...          ...   
9974                               Blackstonian Pencils      18.6900   
9977           Alliance Big Bands Rubber Bands, 12/Pack      13.8600   
9981                                          Avery 476      16.5200   
9982  RSVP Cards & Envelopes, Blank White, 8-1/2" X ...      35.5600   
9988                    Panasonic KX - TS880B Telephone     206.1000   

      Quantidade  
3            5.0  
5            7.0  
9            5.0  
10           9.0  
14           5.0  
...          ...  
9974         7.0  
9977         7.0  
9981         5.0  
9982         7.0  
9988         5.0  

[2128 rows x 11 columns]

In [14]:
# Shape
df[df['Quantidade'].isin([5,7,9,11])].shape

(2128, 11)

In [15]:
df[df['Quantidade'].isin([5,7,9,11])][:100]

ID_Pedido Data_Pedido ID_Cliente     Segmento           Pais  \
3    US-2015-108966  2015-10-11   SO-20335     Consumer  United States   
5    CA-2014-115812  2014-06-09   BH-11710     Consumer  United States   
9    CA-2014-115812  2014-06-09   BH-11710     Consumer  United States   
10   CA-2014-115812  2014-06-09   BH-11710     Consumer  United States   
14   US-2015-118983  2015-11-22   HP-14815  Home Office  United States   
..              ...         ...        ...          ...            ...   
479  CA-2016-100790  2016-06-26   JG-15805    Corporate  United States   
480  CA-2016-100790  2016-06-26   JG-15805    Corporate  United States   
485  CA-2017-140963  2017-06-10   MT-18070  Home Office  United States   
486  CA-2017-140963  2017-06-10   MT-18070  Home Office  United States   
491  CA-2014-113362  2014-09-14   AJ-10960     Consumer  United States   

      Regiao       ID_Produto        Categoria  \
3      South  FUR-TA-10000577        Furniture   
5       West  FUR-FU-10001487        Furniture   
9       West  OFF-AP-10002892  Office Supplies   
10      West  FUR-TA-10001539        Furniture   
14   Central  OFF-AP-10002311  Office Supplies   
..       ...              ...              ...   
479     East  OFF-AR-10003045  Office Supplies   
480     East  OFF-ST-10000689  Office Supplies   
485     West  FUR-BO-10001337        Furniture   
486     West  TEC-PH-10001924       Technology   
491     East  OFF-ST-10001809  Office Supplies   

                                          Nome_Produto  Valor_Venda  \
3        Bretford CR4500 Series Slim Rectangular Table     957.5775   
5    Eldon Expressions Wood and Plastic Desk Access...      48.8600   
9                     Belkin F5C206VTEL 6 Outlet Surge     114.9000   
10            Chromcraft Rectangular Conference Tables    1706.1840   
14   Holmes Replacement Filter for HEPA Air Cleaner...      68.8100   
..                                                 ...          ...   
479                              Prang Colored Pencils      14.7000   
480  Fellowes Strictly Business Drawer File, Letter...     704.2500   
485     O'Sullivan Living Dimensions 2-Shelf Bookcases     514.1650   
486           iHome FM Clock Radio with Lightning Dock     279.9600   
491                  Fellowes Officeware Wire Shelving     449.1500   

     Quantidade  
3           5.0  
5           7.0  
9           5.0  
10          9.0  
14          5.0  
..          ...  
479         5.0  
480         5.0  
485         5.0  
486         5.0  
491         5.0  

[100 rows x 11 columns]